### PR TITLE
[wip] guid metadata records

### PR DESCRIPTION
**Pull request by aaxelb**
## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

### CODE REVIEW

File: api/custom_metadata/views.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            )
+        except osfdb.base.InvalidGuid:
+            raise Http404
+        self.check_object_permissions(self.request, metadata_record)
+        return metadata_record
+
+
+class CustomItemMetadataDetail(JSONAPIBaseView, rest_framework.generics.RetrieveUpdateAPIView):
+    permission_classes = (
+        PermissionWithGetter(ContributorOrPublic, 'guid.referent'),
```
Comment (Johnetordoff) [inline api/custom_metadata/views.py:43]: Jon and I are tending to towards just write an entirely new permission in these cases, it makes it easier it individualize them per view if they need to be changed.

Comment (aaxelb) [reply api/custom_metadata/views.py:43]: huh, i'd have thought to go the other direction, try to make permissions more consistent with fewer permission classes reused more often -- but sure can add bespoke permission here

**Review by brianjgeiger — COMMENTED** [review]
Looking good. Just one implementation question.

File: osf/models/node.py [hunk 2]
```diff
@@ -1807,6 +1812,8 @@ def use_as_template(self, auth, changes=None, top_level=True, parent=None):
         )
         new.save()
 
+        GuidMetadataRecord.objects.copy(from_=self, to_=new)
```
[the remark this answers is not in the document]
Comment (aaxelb) [reply osf/models/node.py:1815]: oh good catch... i feel like we don't want to copy metadata from templates, tho will see what others think (i just got here by searching for `.clone(`)

File: api/base/serializers.py [hunk 3]
```diff
@@ -606,66 +595,53 @@ class RelationshipField(ser.HyperlinkedIdentityField):
     json_api_link = True  # serializes to a links object
 
     def __init__(
-        self, related_view=None, related_view_kwargs=None, self_view=None, self_view_kwargs=None,
-        self_meta=None, related_meta=None, always_embed=False, filter=None, filter_key=None, required=False, **kwargs
+        self, *,
```
Comment (brianjgeiger) [inline api/base/serializers.py:598]: What does this bare asterisk do in the parameter list? Does it collect positional args and then not do anything with them?

Comment (aaxelb) [reply api/base/serializers.py:598]: it puts a cap on positional args -- all the params after `*` *must* be passed as keyword args (the ones before may be positional or keyword)
```
>>> def foo(one, *, two):
...     print(f'{one}, {two}')
... 
>>> foo(1, two=2)
1, 2
>>> foo(one=1, two=2)
1, 2
>>> foo(1, 2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() takes 1 positional argument but 2 were given
```
it gave me comfort in a change like this to guarantee there are no calls anywhere which depend on the order params are listed here

**Comment by aaxelb** [comment]
closing in favor of #10273

**Pull request closed without merging**